### DISCUSSION TITLE
Max loss pct

### DISCUSSION THREAD

**Discussion by ghost** [edtechre/pybroker#146] [General]
Hi, anyone know how this is calculated? I ran a test with risk of 2% of equity per trade with position sizimg based on stop distance and max dollar position size 10% but result shows max loss pct of -7.2%?  Should max 2% unless gap, if so how could I see this and does the backtesr exit at open/close price of bar if below stop for long only as example?

**Comment by ghost** [comment]
Actually if is a daily loss and I have 10 positions open so if several losers this makes sense. But not as a loss per trade. Be good to know how is calculated

**Comment by edtechre** [comment]
Hi @markxbaker,

Which metric are you referring to?

**Reply by edtechre** [reply]
Looks like you are referring to largest_loss_pct. This could have happened in case a bar gapped down below your stop loss value:

```
if low <= self._stop_data[stop.id].value:
     high = price_scope.fetch(stop.symbol, PriceType.HIGH)
     return min(self._stop_data[stop.id].value, high)
```

If the bar's high is below the stop value, then it is used instead of the stop value for the fill price.

You can see the code in the [portfolio module](https://github.com/edtechre/pybroker/blob/master/src/pybroker/portfolio.py#L1279).

**Comment by ghost** [comment]
Thanks Ed, yes that's the metric, understood and thanks! Can I set PriceType.HIGH as OPEN or MID if this occurs? I want a less optimistic exit if possible.Thanks,MarkOn 22 Sept 2024 22:12, Ed West ***@***.***> wrote:
Looks like you are referring to largest_loss_pct. This could have happened in case a bar gapped down below your stop loss value:
if low <= self._stop_data[stop.id].value:
     high = price_scope.fetch(stop.symbol, PriceType.HIGH)
     return min(self._stop_data[stop.id].value, high)

If the bar's high is below the stop value, then it is used instead for the fill price.

**Reply by edtechre** [reply]
Yes, you can use [stop_loss_exit_price](https://www.pybroker.com/en/latest/reference/pybroker.context.html#pybroker.context.ExecContext.stop_loss_exit_price) for this.

**Comment by ghost** [comment]
Thanks Ed.Ed, would this be set for all stop loss prices, or is there a way to set only for this specific scenario when there is a gap through the stop?This way if we exit at the open with slippage(if set) it would create a more realistic exit in the event of a gap through? On 24 Sept 2024 03:28, Ed West ***@***.***> wrote:
Yes, you can use stop_loss_exit_price for this.

**Reply by edtechre** [reply]
You will need to modify portfolio.py here: https://github.com/edtechre/pybroker/blob/master/src/pybroker/portfolio.py#L1261

For example:

```
low = price_scope.fetch(stop.symbol, PriceType.LOW)
if low <= self._stop_data[stop.id].value:
    return price_scope.fetch(stop.symbol, PriceType.OPEN)
```
